apisix/plugins/jwt-auth.lua

```diff
@@ -274,32 +234,16 @@ end
 local function get_rsa_or_ecdsa_keypair(conf, consumer_name)
     local public_key = conf.public_key
     local private_key = conf.private_key
-    -- if keys are present in conf, no need to query vault (fallback)
+
     if public_key and private_key then
         return public_key, private_key
+    elseif public_key and not private_key then
+        return nil, nil, "missing private key"
+    elseif not public_key and private_key then
+        return nil, nil, "missing public key"
+    else
+        return nil, nil, "public and private keys are missing"
```

Comment:
I didn't remove the elseif statements so that a better error message could be returned. Let me know if this is not up to the mark.